packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -450,8 +450,22 @@ function resolveBuiltin(
 ): ResolvedElements {
   const t = resolveTypeElements(ctx, node.typeParameters!.params[0])
   switch (name) {
-    case 'Partial':
-    case 'Required':
+    case 'Partial': {
+      const res: ResolvedElements = { props: {}, calls: t.calls }
+      Object.keys(t.props).forEach(key => {
+        res.props[key] = t.props[key]
+        res.props[key].optional = true
```

Comment:
Should not be mutating the original here - should be cloning with
```ts
res.props[key] = { ...t.props[key], optional: true }
```

Reply:
Done